Scripts/Source/UD_CustomVibratorBase_RenderScript.psc

```diff
@@ -327,7 +327,7 @@ bool Function canVibrate()
 EndFunction
 
 bool Function isVibrating()
-    return _currentVibRemainingDuration != 0 && CurrentVibStrength
+    return _currentVibRemainingDuration > 0 && CurrentVibStrength > 0
```

Comment:
This implementation will not work for vibrators which have infinite duration. Infinite vibration is done by simply specifying the duration as negative. The duration then gets reduces on every update, but it will never reach 0. Is there any specific reason why was it changed ?

Reply:
> Is there any specific reason why was it changed ?

I've encountered a bug where the vibrator turns off but the corresponding icon doesn't update (remaining colored and blinking).

So I randomly corrected all the places that seemed doubtful to me. And I didn’t come across that bug anymore, but this is probably just a coincidence. :)

Thanks for correcting me, I'll check everything again.